scripts/upgrade_cfme.py

```diff
@@ -0,0 +1,216 @@
+        '''
+
+        logger.info("adding repositories to yum")
+        for curr_repo_url in self.repo_list:
+            # Add all the repos to yum repo
+            self.exec_commaned("yum-config-manager "
+                               "--add-repo {repo_url}".format(repo_url=curr_repo_url))
+
+    def update_yum(self):
+        '''
+        initiate upgrading process
+        :return: None
+        '''
+        def monitor_yum():
+            '''
+            give real time monitoring on yum update (give the user experience
+            that the script is still running)
+            :return: None
+            '''
+
+            time.sleep(5)
+            # do as long yum is running on the appliance
+            res = self.exec_commaned(""" ps -ef | grep "yum update" | grep -v grep """,
+                                     ignore_failure=True)
+            while not bool(res.rc):
+                logger.info("yum still running")
+                time.sleep(10)
+                res = self.exec_commaned(""" ps -ef | grep "yum update" | grep -v grep """,
+                                         ignore_failure=True)
+
+        # start thr monitoring thread
+        monitor = threading.Thread(target=monitor_yum)
+        monitor.start()
+
+        logger.info("initiating yum update")
+        self.ssh_client.run_command("yum update -y --nogpgcheck")
+
+    def stop_cfme(self):
+        '''
+        stop CFME engine process
+        :return:
+        '''
+        logger.info("stopping the cfme engine (evmserverd)")
+
+        result = self.exec_commaned("systemctl status evmserverd | "
+                                    "grep Active: | awk {'print $2'}", ignore_failure=True)
+
+        if "active" in result.output:
+            self.exec_commaned("systemctl stop evmserverd")
+
+    def start_cfme(self):
+        '''
+        start CFME engine process
+        :return: None
+        '''
+        logger.info("starting the cfme engine (evmserverd)")
+        result = self.exec_commaned("systemctl status evmserverd | grep Active: | awk {'print $2'}",
+                                    ignore_failure=True)
+
+        if "inactive" in result.output:
+            self.exec_commaned("systemctl start evmserverd")
+
+    def yum_register(self, username, password):
+        '''
+        registaer to subscription manager
+        :param username: username to sign with
+        :param password: password to sign with
+        :return: None
+        '''
+        self.exec_commaned("subscription-manager register --username={username} "
+                           "--password={password} "
+                           "--force".format(username=username, password=password))
+
+    def validate_cfme_version(self):
+        '''
+        Validating if the cfme version  is fit
+        :return:
+        '''
+        logger.info("waiting for 120 seconds for main process to load "
+                    "before running version validation")
+        time.sleep(120)
```

Comment:
Don't you want to use wait_for and wait until the api starts responding instead?